### DISCUSSION TITLE
Быстрая форма и передача информацию в нее

### DISCUSSION THREAD

**Discussion by k1erk** [totumonline/totum-ru-issues-and-discussions#242] [Есть идея, а что если...]
Всем доброго времени суток. Есть ли возможность во время загрузки быстрой формы вставленной через тег script передать данные в input, которые получаем со стороннего сайта. На стороннем сайте вставлена форма и доступна она только авторизованным, в теле DOM document присутствует Имя пользователя, который сейчас залогинен, есть желание его вставить в input в форме. Может есть идеи как это сделать. Jquery не видит форму.

**Comment by ttm-alexeypolunin** [comment]
Такое есть для Форм, ща посмотрим, что у нас там для Быстрых форм

**Comment by k1erk** [comment]
> Всем доброго времени суток. Есть ли возможность во время загрузки быстрой формы вставленной через тег script передать данные в input, которые получаем со стороннего сайта. На стороннем сайте вставлена форма и доступна она только авторизованным, в теле DOM document присутствует Имя пользователя, который сейчас залогинен, есть желание его вставить в input в форме. Может есть идеи как это сделать. Jquery не видит форму.

UPD:
получилось поставить через jquery ('.ttm-fieldCell[data-field-name="imya_sayt"] input').attr('value', trim_name); 
Значение подставляется и его видно через просмотр кода 
<input aria-invalid="false" disabled="" type="string" class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense" value="tst">
Но именно это поле прилетает пустым в тотум. 
Может будут мысли с чем связано?

**Reply by ttm-alexeypolunin** [reply]
Собственно быстрая форма не предназначена для получения get/post внутрь себя. Мы тоже пришли к выводу, что тут возможна простановка значения через JS в техническое поле и его скрытие от пользователя.

Скорее всего эта операция воткнется в CORS и нужно прописать разрешение в настройках nginx.

Мы поэксперементируем с этим на следующей неделе.

**Reply by ttm-alexeypolunin** [reply]
Еще у нас есть мысль про вьютайп JS для текстового поля, что бы можно было в форму и быструю форму совать кастомные скрипты со стороны Totum.

**Reply by ttm-alexeypolunin** [reply]
Чтот там реакт не хочет брать никаким боком ничего из input. Как-то это нам надо выставить функцию наружу, чтоб можно было записать.

**Reply by k1erk** [reply]
> Чтот там реакт не хочет брать никаким боком ничего из input. Как-то это нам надо выставить функцию наружу, чтоб можно было записать.

да, я уже пытался и post запрос сделать на отправку в скрипт, но он не видит метод ((

**Reply by ttm-alexeypolunin** [reply]
В последнем обновлении `4.8.50.1` есть `viewtype = js` для поля тип `data`.

Оно должно быть доступно для роли от которой вызывается форма, иначе оно не отобразится в DOM. Для пользователя оно визуально скрывается. 

<img width="789" alt="Снимок экрана 2022-11-25 в 18 15 40" src="https://user-images.githubusercontent.com/55755565/204014173-ae6ded67-0dc7-4c83-97e3-1b3069225f5d.png">

В параметр `id` задаете свой `id` по которому можно обратится к этому полю в DOM.

Что бы передать ему данные надо:

```
<script>
        let interv = setInterval(()=>{
            let element =  document.getElementById('your_id_for_field');
            if(element && element.setTotumValue){
                clearInterval(interv)
                element.setTotumValue(JSON.stringify(
                    ["your_data"]));
            }
        }, 100);
    </script>
```

**Comment by k1erk** [comment]
Спасибо, попробуем.
